doc/source/train/examples/pytorch/distributing-pytorch/README.ipynb

```diff
@@ -0,0 +1,599 @@
+    "\n",
+    "![Train Run](images/train_run.png)\n",
+    "\n",
+    "Clicking into an individual worker brings you to a more detailed worker page. \n",
+    "\n",
+    "![Train Worker](images/train_worker.png)\n",
+    "\n",
+    "If your worker is still alive, you can click the \"CPU flame graph,\" \"stack trace,\" or \"memory profiling\" links in the overall run page or the individual worker page. Clicking the \"CPU flame graph\" link profiles your run with py-spy for 5 seconds and show you a CPU flame graph. Clicking the \"stack trace\" link shows you the current stack trace of your job - this is useful for debugging hanging jobs. Finally, clicking \"memory profiling\" profiles your run with memray for 5 seconds and show you a memory flame graph.\n",
+    "\n",
+    "You can also click the \"Metrics\" tab on the navigation bar to view useful stats about your cluster, such as GPU utilization and metrics about ray [actors](https://docs.ray.io/en/latest/ray-core/actors.html) and [tasks](https://docs.ray.io/en/latest/ray-core/tasks.html).\n",
+    "\n",
+    "![Metrics Dashboard](images/metrics_dashboard.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Scale data ingest separately from training with Ray Data\n",
+    "\n",
+    "\n",
+    "Let’s modify this example to load data with Ray Data instead of the native PyTorch DataLoader. With just a few modifications, you can scale data preprocessing and training separately; for example, you can do the former with a pool of CPU workers and the latter with a pool of GPU workers. See [here](https://docs.ray.io/en/latest/data/comparisons.html#how-does-ray-data-compare-to-other-solutions-for-ml-training-ingest) for a comparison between Ray Data and PyTorch data loading.\n",
+    "\n",
+    "First, create a [Ray Data Dataset](https://docs.ray.io/en/latest/data/key-concepts.html#datasets-and-blocks). Note the following about the code below:\n",
+    "* You are now loading the data from the [Ansycale default storage bucket](https://docs.anyscale.com/configuration/storage/#anyscale-default-storage-bucket) instead of from PyTorch directly. \n",
+    "* Each row of a Ray Data Dataset is a dict so you need to access its contents by key.\n",
+    "* Ray Data loads and transforms data lazily, which means that only the training workers materialize the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "STORAGE_PATH_PREFIX = os.environ.get(\"ANYSCALE_ARTIFACT_STORAGE\", \"artifact_storage\")\n",
+    "STORAGE_PATH = f\"{STORAGE_PATH_PREFIX}/ray_summit_24_train_demo\"\n",
+    "\n",
+    "def transform_cifar(row: dict):\n",
+    "    # Define the torchvision transform.\n",
+    "    transform = transforms.Compose([ToTensor(), Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])\n",
+    "    row[\"image\"] = transform(row[\"image\"])\n",
+    "    return row\n",
+    "\n",
```

Comment:
Let's show the `ds.schema()` in an earlier cell so that people understand how we figured out these dict keys.

also better typehints

```suggestion
    "def transform_cifar(row: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
    "    # Define the torchvision transform.\n",
    "    transform = transforms.Compose([ToTensor(), Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])\n",
    "    row[\"image\"] = transform(row[\"image\"])\n",
    "    return row\n",
    "\n",
```

Reply:
done.